eng/restore/optimizationData.targets

```diff
@@ -1,11 +1,11 @@
 ﻿<Project>
 
   <ItemGroup>
-    <IBCPackage Include="$(WindowsCoreFxOptimizationDataPackage)" Version="$(optimizationwindows_ntx64IBCCoreFxVersion)" />
-    <IBCPackage Include="$(LinuxCoreFxOptimizationDataPackage)" Version="$(optimizationlinuxx64IBCCoreFxVersion)" />
-    <IBCPackage Include="$(WindowsOptimizationIBCCoreCLRPackage)" Version="$(optimizationwindows_ntx64IBCCoreCLRVersion)" />
-    <IBCPackage Include="$(LinuxOptimizationIBCCoreCLRPackage)" Version="$(optimizationlinuxx64IBCCoreCLRVersion)" />
-    <PackageReference Include="@(IBCPackage)" PrivateAssets="all" IsImplicitlyDefined="true" />
+    <IBCPackage Include="$(WindowsCoreFxOptimizationDataPackage)" Version="[$(optimizationwindows_ntx64IBCCoreFxVersion)]" />
+    <IBCPackage Include="$(LinuxCoreFxOptimizationDataPackage)" Version="[$(optimizationlinuxx64IBCCoreFxVersion)]" />
+    <IBCPackage Include="$(WindowsOptimizationIBCCoreCLRPackage)" Version="[$(optimizationwindows_ntx64IBCCoreCLRVersion)]" />
+    <IBCPackage Include="$(LinuxOptimizationIBCCoreCLRPackage)" Version="[$(optimizationlinuxx64IBCCoreCLRVersion)]" />
```

Comment:
Should these be conditioned depending on the OS that we're building? @ericstj I don't know exactly how this data is used, but my understanding tells me that we only need the one for the OS we're building?

Reply:
I'm not changing how this works, I'm simply switching from PackageReference to PackageDownload here. This PR is already complex enough, would you mind if we tackle this in a follow-up PR? I don't want to introduce additional unnecessary risk.

Reply:
> I'm not changing how this works, I'm simply switching from PackageReference to PackageDownload here.

Yeah, I understand that 😄, just asking if it is necessary to download all of them always. If @ericstj confirms that we can condition these, then we should open an issue for that, I'm fine with doing it in a follow-up PR.

Reply:
Let's not touch it, I actually see some room for improvement here that I'll follow up on offline.

Reply:
https://github.com/dotnet/runtime/issues/34422